packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -88,6 +89,65 @@ export const onSingleSelectHandler = <T extends Record<string, unknown>>(
   return newState;
 };
 
+/**
+ * Handles selection when clicked on a row while shift key is pressed.
+ */
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance?: TableInstance<T>,
+  onSelect?: (
+    selectedData: T[] | undefined,
+    tableState?: TableState<T>,
+  ) => void,
+  isRowDisabled?: (rowData: T) => boolean,
+) => {
+  let startIndex = instance?.flatRows.findIndex(
+    (row) => row.id === state.lastSelectedRow,
+  );
+  let endIndex = instance?.flatRows.findIndex((row) => row.id === action.id);
+
+  if (startIndex == null || startIndex < 0) {
+    startIndex = 0;
+  }
+  if (endIndex == null || endIndex < 0) {
+    endIndex = 0;
+  }
+
+  if (startIndex > endIndex) {
+    const temp = startIndex;
+    startIndex = endIndex;
+    endIndex = temp;
+  }
+
+  const selectedRowIds: Record<string, boolean> = {};
+
+  // 1. Select all rows between start and end
+  instance?.flatRows
+    .slice(startIndex, endIndex + 1)
+    .forEach((r) => (selectedRowIds[r.id] = true));
+
+  // 2. Select all children of the last row (endIndex)
+  if (instance != null) {
+    const handleRow = (row: Row<T>) => {
+      selectedRowIds[row.id] = true;
+      row.subRows.forEach((r) => handleRow(r));
+    };
+    handleRow(instance.rowsById[action.id]);
```

Comment:
What happens when user clicks row 5, then row 2, startIndex > endIndex? I think you should pass `endIndex` here and not action.id.

Reply:
**[DONE]** Ohh, nice catch, thanks